### ISSUE TITLE
task: re-create format_version.txt on recovery

### ISSUE THREAD

**Issue by harlinb** [yandex/ClickHouse#1569]
When using a custom partitioned ReplicatedMergeTree 
as in https://github.com/yandex/ClickHouse/blob/master/dbms/tests/queries/0_stateless/00502_custom_partitioning_local.sql

The data directory "data/\<database\>/\<table\>" requires a file "format_version.txt" with the value 1 in it.
If this directory is lost, in the event of a disk failure for example, clickhouse will not add the format_version.txt file and you will see the following error:

``` error: DB::Exception: MergeTree data format version on disk doesn't support custom partitioning, stack trace:``` 

as a work around you can simply
``` printf 1 > data/<database>/<table>/format_version.txt```

It would speed up recovery and remove some manual intervention if clickhouse could deduce the format version of the table and create the file itself.

**Comment by ivankosianenko** [comment]
Hi @harlinb 
We are created format_version.txt
but I didn't help us:
<Error> Application: DB::Exception: Cannot create table from metadata file /opt/clickhouse/metadata/af_0//events_partition.sql, error: DB::Exception: Bad version file:
/opt/clickhouse//data/af_0/events_partition/format_version.txt, stack trace:

**Comment by ztlpn** [comment]
The problem is probably the trailing line break. Re-create it with `echo -n 1 > /opt/clickhouse/metadata/af_0/events_partition.sql`.

I should fix it properly.

**Comment by wawanawna** [comment]
We also have a problems with this issue.
We have MaterializedView which we map to RAM (dir with MV table)
And each time after restart we see

> 2018.07.04 09:51:36.334149 [ 1 ] <Error> Application: DB::Exception: Cannot create table from metadata file /mnt/local/clickhouse/metadata/default//XXX.sql, error: DB::Exception: MergeTree data format version on disk doesn't support custom partitioning, stack trace:
> 0. /usr/bin/clickhouse-server(StackTrace::StackTrace()+0x15) [0x84eb125]

We use *ENGINE = MergeTree* for MV

Workaround with format_version.txt helped us, but will be nice to have a fix if possible.


PS: Thx for a great product, really awesome

**Comment by hakman** [comment]
@wawanawna and me found the above workaround to be working quite well. We used tmpfiles.d to recreate _format_version.txt_ at startup:
```
> cat /etc/tmpfiles.d/clickhouse.conf
/mnt/local/clickhouse/data/default/table/format_version.txt  0644  clickhouse  clickhouse  -  1
```

Also, tried to create a more permanent fix, so this is the reason for #2593.
